src/test/java/co/rsk/federate/btcreleaseclient/cache/PegoutSignedCacheFactoryTest.java

```diff
@@ -0,0 +1,40 @@
+package co.rsk.federate.btcreleaseclient.cache;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertSame;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import java.time.Duration;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.NullSource;
+import org.junit.jupiter.params.provider.ValueSource;
+
+class PegoutSignedCacheFactoryTest {
+
+  @ParameterizedTest
+  @NullSource
+  @ValueSource(longs = { -10, 0 })
+  void getInstance_shouldThrowIllegalArgumentException_whenTtlIsInvalid(Long ttl) {
+    Duration invalidTtl = ttl != null ? Duration.ofMinutes(ttl) : null;
+    String expectedErrorMessage = String.format(
+        "Invalid pegouts signed cache TTL value in minutes supplied: %s", ttl);
+
+    IllegalArgumentException exception = assertThrows(IllegalArgumentException.class,
+        () -> PegoutSignedCacheFactory.getInstance(invalidTtl));
+    assertEquals(expectedErrorMessage, exception.getMessage());
+  }
+
+  @Test
+  void getInstance_shouldReturnSameInstance_whenCalledMultipleTimesWithValidTtl() {
+    Duration ttl = Duration.ofMinutes(10);
```

Comment:
So what happens if you call `getInstance` with different ttl values? Is it still the same instance? Which ttl value is used?

Reply:
Yes, once it is instantiated it returns the same instance as the first one. The ttl will be passed from a config file which is not known at creation time. If we want it to be parameterless then we can include a setter or init method that updates the ttl value for `PegoutSignedCacheImpl`.

Reply:
From what we discussed offline, we will not be using the singleton patter for the cache impl.